packages/client/src/rpc/modules/debug.ts

```diff
@@ -278,4 +288,80 @@ export class Debug {
       message: err.message.toString(),
     }
   }
+
+  /**
+   * Returns a limited set of storage keys belonging to an account.
+   * @param params An array of 5 parameters:
+   *    1. The hash of the block that contains the requested account storage.
+   *    2. The number index of the requested transaction within the block.
+   *    3. The address of the account.
+   *    4. The smallest (hashed) storage key that will be returned. To include the entire range, pass '0x00'.
+   *    5. The maximum number of storage values that will be returned.
+   * @returns A {@link StorageRange} object that will contain at most `limit` entries in its `storage` field.
+   * The object will also contain `nextKey`, the next (hashed) storage key after the range included in `storage`.
+   */
+  async storageRangeAt(params: [string, number, string, string, number]) {
+    const [blockHash, txIndex, account, startKey, limit] = params
+
+    if (txIndex < 0) {
+      throw {
+        code: INTERNAL_ERROR,
+        message: 'txIndex cannot be smaller than 0.',
+      }
+    }
+
+    let address: Address
+    let blockHashBytes: Uint8Array
+    let startKeyBytes: Uint8Array
+    try {
+      address = Address.fromString(account)
+      blockHashBytes = hexToBytes(blockHash)
+      startKeyBytes = hexToBytes(startKey)
+    } catch (err: any) {
+      throw {
+        code: INVALID_PARAMS,
+        message: err.message.toString(),
+      }
+    }
+
+    let block: Block
+    try {
+      block = await this.service.chain.getBlock(blockHashBytes)
+    } catch (err: any) {
+      throw {
+        code: INTERNAL_ERROR,
+        message: 'Could not get requested block hash.',
+      }
+    }
+
+    if (txIndex + 1 > block.transactions.length) {
```

Comment:
```suggestion
    if (txIndex  >= block.transactions.length) {
```
more comprehensible about the check